x/cork/keeper/keeper.go

```diff
@@ -222,6 +303,56 @@ func (k Keeper) GetApprovedCorks(ctx sdk.Context, threshold sdk.Dec) (approvedCo
 	return winningCorks
 }
 
+func (k Keeper) GetApprovedScheduledCorks(ctx sdk.Context, currentBlockHeight uint64, threshold sdk.Dec) (approvedCorks []types.Cork) {
+
+	corksForBlockHeight := make(map[uint64][]types.Cork)
+	var corkPowersForBlockHeight map[uint64][]int64
+
+	totalPower := k.stakingKeeper.GetLastTotalPower(ctx)
+
+	k.IterateScheduledCorks(ctx, func(val sdk.ValAddress, scheduledBlockHeight uint64, addr common.Address, cork types.Cork) (stop bool) {
+		// only operate on scheduled corksForBlockHeight that are valid
+		if currentBlockHeight < scheduledBlockHeight {
+			return false
+		}
+
+		validator := k.stakingKeeper.Validator(ctx, val)
+		validatorPower := validator.GetConsensusPower(k.stakingKeeper.PowerReduction(ctx))
+
+		found := false
+		for i, c := range corksForBlockHeight[scheduledBlockHeight] {
+			if c.Equals(cork) {
+				corkPowersForBlockHeight[scheduledBlockHeight][i] += validatorPower
```

Comment:
So, `corkPowersForBlockHeight` is a map of a block height to a slice of int64 objects. This `+=` assignment is using the position of the cork in `corksForBlockHeight` as an index into that array, but when it is created below it appends the first power as a single element at index zero in the array. This means that an access for any "found" cork whose index is not zero in the `corksForBlockHeight` value would be an out-of-bounds read. This could either be fixed by using an append here to store all the powers and summing them later or by using a single int64 (or uint64) to progressively sum them.